### PR TITLE
feat: implemented extension settings

### PR DESCRIPTION
**Pull request by jbristowe**

### CODE REVIEW

**Review by benPearce1 — APPROVED** [review]
Reviewed over zoom. LGTM

**Review by benPearce1 — APPROVED** [review]
reviewed over zoom. LGTM 👍

**Review by benPearce1 — APPROVED** [review]
reviewed over zoom. LGTM 👍

**Review by slewis74 — COMMENTED** [review]
Looking good. I haven't tried testing locally.

I think it could be worth adding the allowed values for `environment_type` to the docs.

There are also a number of checks compaining about comment format in the code (e.g. comment has a method name that isn't the actual method name)

File: docs/data-sources/environments.md [hunk 1]
```diff
@@ -46,8 +46,36 @@ Read-Only:
...
 - `sort_order` (Number) The order number to sort an environment.
+- `space_id` (String) The space ID associated with this environment.
 - `use_guided_failure` (Boolean)
 
+<a id="nestedatt--environments--jira_extension_settings"></a>
+### Nested Schema for `environments.jira_extension_settings`
+
+Read-Only:
+
+- `environment_type` (String)
```
[the remark this answers is not in the document]
Comment (jbristowe) [reply docs/data-sources/environments.md:63]: These values are listed in the description for the Terraform schema. Unfortunately, it looks like the [documentation generation tool we use](https://github.com/hashicorp/terraform-plugin-docs) may have a bug when it comes to emitting nested schemas: [Descriptions for nested schemas are missing (inconsistent behaviour) #28](https://github.com/hashicorp/terraform-plugin-docs/issues/28).

**Review by domenicsim1 — APPROVED** [review]

**Pull request merged**